### PR TITLE
support configure some hyper params

### PR DESCRIPTION
**Pull request by typhoonzero**
Support some hyper parameters, so can train set below params when training like:

```sql
SELECT *
FROM iris.train
TRAIN DNNClassifier
WITH n_classes = 3, hidden_units = [10, 20], BATCH_SIZE = 128, STEPS = 100, EPOCHS=30
COLUMN sepal_length, sepal_width, petal_length, petal_width
LABEL class
INTO sqlflow_models.my_dnn_model;
```

### CODE REVIEW

File: cmd/sqlflowserver/main_test.go [hunk 1]
```diff
@@ -355,6 +357,7 @@ FROM iris.predict LIMIT 5;`
 // CaseTrainTextClassification is a simple End-to-End testing for case training
 // text classification models.
 func CaseTrainTextClassification(t *testing.T) {
+	t.Skip()
```
Comment (tonyyang-svail) [inline cmd/sqlflowserver/main_test.go]: What is the purpose of this skip? Is `CaseTrainTextClassification` time consuming?

File: sql/codegen.go [hunk 2]
```diff
@@ -200,13 +200,32 @@ import logging
...
+STEPS = 1000
+EPOCHS = 1
 NUM_BUCKETS=160000
 EMBEDDING_WIDTH=128
 
+train_args = dict()
+{{range $key, $value := .Attrs}}
+{{if eq $key "BATCHSIZE"}}
+BATCHSIZE = {{$value}}
+{{else if eq $key "BATCH_SIZE"}}
```
Comment (tonyyang-svail) [inline sql/codegen.go]: Not sure we need to support both `BATCHSIZE` and `BATCH_SIZE`..

File: sql/codegen.go [hunk 3]
```diff
@@ -290,7 +309,7 @@ eval_result = classifier.evaluate(eval_input_fn(X, Y, BATCHSIZE), verbose=0)
 print("Training set accuracy: {accuracy:0.5f}".format(**{"accuracy": eval_result[1]}))
 {{else}}
 eval_result = classifier.evaluate(
-	input_fn=lambda:eval_input_fn(X, Y, BATCHSIZE), steps=STEP)
+	input_fn=lambda:eval_input_fn(X, Y, BATCHSIZE), steps=STEPS)
```
Comment (tonyyang-svail) [inline sql/codegen.go:308]: What if `STEPS` is larger than the eval dataset batch numbers?

I don't think `steps` should be configurable. A `PREDICT` should always go through the whole selected data. If a user wants a subset of the select, (s)he can put the restriction on the StandardSelect
```sql
SELECT ...
WHERE ...
LIMIT ...
PREDICT...
```

File: sql/codegen.go [hunk 4]
```diff
@@ -200,13 +200,32 @@ import logging
...
 EMBEDDING_WIDTH=128
 
+train_args = dict()
+{{range $key, $value := .Attrs}}
+{{if eq $key "BATCHSIZE"}}
+BATCHSIZE = {{$value}}
+{{else if eq $key "BATCH_SIZE"}}
+BATCHSIZE = {{$value}}
+{{else if eq $key "EPOCHS"}}
+EPOCHES = {{$value}}
```
Comment (tonyyang-svail) [inline sql/codegen.go]: `EPOCHES` or `EPOCHS`?

File: sql/codegen.go [hunk 5]
```diff
@@ -271,13 +290,13 @@ classifier.compile(optimizer=classifier.default_optimizer(),
 	loss=classifier.default_loss(),
 	metrics=["accuracy"])
 classifier.fit(train_input_fn(X, Y, BATCHSIZE),
-	epochs=classifier.default_training_epochs(),
-	steps_per_epoch=STEP, verbose=0)
+	epochs=EPOCHS if EPOCHS != classifier.default_training_epochs() else classifier.default_training_epochs(),
```
Comment (tonyyang-svail) [inline sql/codegen.go]: In the current configuration, I think `classifier.default_training_epochs()` will never be used...  Given `EPOCHS` has been initialized to a number, this line is equivalent to
```python
epochs = EPOCHS
```

How about we rewrite line 208 to `EPOCHS = None` then rewrite here as
```python
epochs=EPOCHS if EPOCHS else classifier.default_training_epochs()
```
So the logic becomes
```
if user_defined:
    epochs = user_defined_epochs
else:
    epochs = model_default_epochs
```

File: sql/codegen.go (hunk 3 above)
Comment (typhoonzero) [reply sql/codegen.go:308]: Determine the actual dataset size ( total number of elements ) sometimes is time-consuming especially on very large datasets ( like dataset stored in Hive ), if user didn't provide `LIMIT` clause then we must execute `SELECT count(*)  FROM table WHERE ...` then calculate `STEPS`.

I'm not sure whether we should do this now, so put `STEPS` configurable for now and move on to an automatic solution.

File: cmd/sqlflowserver/main_test.go (hunk 1 above)
Comment (typhoonzero) [reply cmd/sqlflowserver/main_test.go]: Just for test, removed after clean.

File: sql/codegen.go (hunk 2 above)
Comment (typhoonzero) [reply sql/codegen.go]: Done

File: sql/codegen.go (hunk 4 above)
Comment (typhoonzero) [reply sql/codegen.go]: Done

File: sql/codegen.go (hunk 5 above)
Comment (typhoonzero) [reply sql/codegen.go]: Thanks, that's clever. Done!

File: sql/codegen.go (hunk 3 above)
Comment (tonyyang-svail) [reply sql/codegen.go:308]: We don't need to determine the actual dataset size. If we don't specify `steps`, `classifier.evaluate` will consume all the data points returned by`eval_input_fn`.

> steps: Integer or None. Total number of steps (batches of samples) before declaring the evaluation round finished. Ignored with the default value of None.

**Review by tonyyang-svail — APPROVED** [review]

File: sql/codegen.go (hunk 3 above)
Comment (typhoonzero) [reply sql/codegen.go:308]: Checked keras's document: https://keras.io/models/model/#fit `model.fit` can accept `steps_per_epoch=None` also. Removing this in the next PR while I'll refine the data reading part

**Pull request merged**